### PR TITLE
ci: testing with `torch==2.3` /test and move `2.4` /nightly

### PR DESCRIPTION
**Pull request by Borda**
<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?

As 2.3 branch was already cut https://github.com/pytorch/pytorch/tree/release/2.3
and Meta is [building dev versions](https://download.pytorch.org/whl/test/torch/); we will switch to using proper versions

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: dockers/ubuntu-cuda/Dockerfile [hunk 1]
```diff
@@ -136,7 +136,14 @@ RUN \
         cd .. && \
         rm -rf Fuser ; \
     elif [ "${TORCH_INSTALL}" == "test" ]; then \
-        echo "Not supported option" ; \
+        # building nvFuser from source
```
Comment (Borda) [inline dockers/ubuntu-cuda/Dockerfile:139]: @mruberry, which version of nvFuser shall we use for the test version of Torch 2.3
for PT 2.2 there is PyPI version but none was yet build for `nvfuser-cu118-torch23`

Comment (tfogal) [reply dockers/ubuntu-cuda/Dockerfile:139]: I think this is actually more a @jjsjann123 question

Comment (Borda) [reply dockers/ubuntu-cuda/Dockerfile:141]: I think it is fine for now to compile it from source...

**Review by t-vi — APPROVED** [review]
Seems good, thannk you @Borda

**Pull request merged**